force-app/main/default/classes/SOQL.cls

```diff
@@ -130,9 +153,11 @@ public with sharing class SOQL {
      * @return A list of Lead records.
      * Update the corresponding test class and testGetLeadsBySourceOrTitle method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}    
+    public static List<Lead> getLeadsBySourceOrTitle(String leadSource, String keyword) {
+        
+        List<Lead> leadsToReturn = [SELECT Id, LastName, FirstName, Title, LeadSource, Company, Phone, Email FROM Lead WHERE LeadSource LIKE :'%' + leadSource + '%' OR Title LIKE :'%' + keyword + '%'];
```

Comment:
LeadSource should be an exact match while Title should just contain the keyword (like you've done)